staging/src/k8s.io/apimachinery/pkg/conversion/queryparams/convert.go

```diff
@@ -90,7 +90,14 @@ func customMarshalValue(value reflect.Value) (reflect.Value, bool) {
 
 	marshaler, ok := value.Interface().(Marshaler)
 	if !ok {
-		return reflect.Value{}, false
+		if !isPointerKind(value.Kind()) && value.CanAddr() {
```

Comment:
No, the case where `Marshaler` interface is implemented by a type with a non-pointer receiver of the method is handled on line 91. Added a test - please correct me if I misunderstood you and this is not the test you meant.